File: code_samples/scala/utilities_regex.scala
/**
 * Utilities: Regex
 * AI/ML Training Sample
 */

class Regex {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object RegexExample {
  def main(args: Array[String]): Unit = {
    val instance = new Regex()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}

File: code_samples/haskell/utilities_regex.hs
-- Utilities: Regex
-- AI/ML Training Sample

module Regex where

data Regex = Regex {
    getData :: String
} deriving (Show, Eq)

process :: Regex -> String -> Regex
process obj input = obj { getData = input }

validate :: Regex -> Bool
validate obj = not (null (getData obj))

main :: IO ()
main = do
    let instance = Regex { getData = "" }
    let updated = process instance "example"
    putStrLn $ "Data: " ++ getData updated
    putStrLn $ "Valid: " ++ show (validate updated)
